### PR TITLE
Refactor Primitive

### PR DESCRIPTION
**Pull request by MellKam**
#269

### CODE REVIEW

File: packages/radix-vue/src/ScrollArea/ScrollAreaScrollbarImpl.vue [hunk 1]
```diff
@@ -128,5 +130,5 @@ useResizeObserver(rootContext?.content, handleSizeChange);
     @pointerup="handlePointerUp"
```
Comment (zernonia) [inline packages/radix-vue/src/ScrollArea/ScrollAreaScrollbarImpl.vue:130]: we need to add `as` as injectedValue in `SCROLL_AREA_SCROLLBAR_INJECTION_KEY`.

then we can pass in the correct `:as="scrollbarContext?.as"` here

**Review by zernonia — COMMENTED** [review]
Teriffic work @MellKam !


Could you pull in latest main? I've added a few internal component, which is using `PrimitiveDiv` still, which is causing error.

**Comment by MellKam** [comment]
> Teriffic work @MellKam !
> 
> Could you pull in latest main? I've added a few internal component, which is using `PrimitiveDiv` still, which is causing error.

Thanks. I'll try to do it soon.

File: packages/radix-vue/src/ScrollArea/ScrollAreaScrollbarImpl.vue (hunk 1 above)
Comment (MellKam) [reply packages/radix-vue/src/ScrollArea/ScrollAreaScrollbarImpl.vue:131]: [178e90ab59f8a4c659e2a600e0e31e09f06369f7](https://github.com/radix-vue/radix-vue/commit/178e90ab59f8a4c659e2a600e0e31e09f06369f7)

**Review by zernonia — APPROVED** [review]
nicely done @MellKam ! 🤩

**Pull request merged**